index.html

```diff
@@ -1,24 +1,76 @@
+<!DOCTYPE html>
 <html>
   <head>
     <meta charset="utf-8">
     <link rel="stylesheet" href="pico.min.css">
     <link rel="stylesheet" href="rkiformulare.css">
+    <script src="https://unpkg.com/html5-qrcode" type="text/javascript"></script>
     <title>qr scanner</title>
-
   </head>
   <body class="container">
     <h1>RKI Formulare</h1>
     <div class="grid">
-      <button onclick="openCamera()">QR-Code scannen</button>
+      <button>QR-Code scannen</button>
       <div>
         <label for="BrowseImage"><span id="label_link" role="button">QR-Code aus Foto lesen</span></label>
-        <Input hidden type="file" id="BrowseImage" accept="image/*" name="pictureToScan">
+        <input hidden type="file" id="BrowseImage" accept="image/*" name="pictureToScan">
        </div>
     </div>
-    <script>
+    <div id="link"></div>
+    
+    <div id="reader" width="600px"></div>
+    <select id="cameraBox" hidden></select>
+    <script type="module">
+      import {parseVcard, fillSelectionBox} from './rkiformulare.js';
+        // This method will trigger user permissions
+      let cameraId;
+      let cameras;
+      let cameralabels = [];
+      Html5Qrcode.getCameras().then(devices => {
+        /**
+         * devices would be an array of objects of type:
+         * { id: "id", label: "label" }
+         */
+        if (devices && devices.length) {
+          cameraId = devices[0];
+          cameras = devices;
+          for(let i = 0; i < cameras.length; i++){
+            cameralabels[i] = cameras[i].label;
+          }
+          console.log(cameraId);
+          console.log(cameras);
+          console.log(cameralabels);
+          // .. use this to start scanning.
+        }
+      }).catch(err => {
+        // handle err
+      });
+      
       function openCamera() {
-        alert("scanning...");
-      }
+        const html5QrCode = new Html5Qrcode("reader");
+        document.getElementById("cameraBox").hidden = false;
+        const qrCodeSuccessCallback = (decodedText, decodedResult) => {
+         html5QrCode.stop();
+         document.getElementById("cameraBox").hidden = true;
+         let RKIlink = document.createElement("a");
+         let user = parseVcard(decodedText);
+         const d = new Date();
+         RKIlink.href = 'https://rkiformulare.online/rkiformulare.html?firstName=' + user.firstName + '&secondName='+user.secondName+'&birthdate='+user.birthDay+'&address='+user.street+'&date='+d.getDate()+"."+(d.getMonth()+1)+"."+d.getFullYear() + "&openPrintDialog=true";
```

Comment:
hier bitte eine relative Adresse nutzen, also statt `https://rkiformulare.online/rkiformulare.html` einfach `./rkiformulare.html` ... dann wird in preview in bei dem lokalen testen auch die preview bzw. lokale Version von `rkiformulare.html` genutzt